drivers/storage/portworx/component/telemetry_java.go

```diff
@@ -144,16 +147,30 @@ func (t *telemetry) deployMetricsCollector(
 		return nil
 	}
 
-	if len(cluster.Status.ClusterUID) == 0 {
-		logrus.Warn("clusterUID is empty, wait for it to fill collector proxy config")
-		return nil
-	}
-
-	if err := t.createCollectorServiceAccount(cluster.Namespace, ownerRef); err != nil {
-		return err
+	if t.isCCMGoSupported {
+		// Run CCM Go, use shared service account, cluster role and cluster role binding
+		if err := k8sutil.DeleteServiceAccount(t.k8sClient, CollectorServiceAccountName, cluster.Namespace, *ownerRef); err != nil {
+			return err
+		}
+		if err := k8sutil.DeleteClusterRole(t.k8sClient, CollectorClusterRoleName); err != nil {
+			return err
+		}
+		if err := k8sutil.DeleteClusterRoleBinding(t.k8sClient, CollectorClusterRoleBindingName); err != nil {
+			return err
+		}
+	} else {
+		if err := t.createCollectorServiceAccount(cluster.Namespace, ownerRef); err != nil {
+			return err
+		}
+		if err := t.createCollectorClusterRole(); err != nil {
+			return err
+		}
+		if err := t.createCollectorClusterRoleBinding(cluster.Namespace); err != nil {
```

Comment:
how about role and roleBinding?

Reply:
They will not be changed, still same as before but service account will point to the new one.

Reply:
clusterRole and role should coexist. they should be reused/created/deleted at the same places.

Reply:
After upgrade the cluster role is used by registration, log uploaded and metrics collector roles, I’ll group the object creation based on components like metrics collector later. Here is when upgrading, we delete the old service account, cluster role and binding, reconcile it to use the new one which should be created already at this point